### PR TITLE
Add ASN1NumericString support

### PR DESCRIPTION
**Pull request by cyberbobs**

### CODE REVIEW

File: lib/asn1numericstring.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  ///
+  /// Create an [ASN1NumericString] initialized with String value.
+  ///
+  /// Optionally override the tag
+  ///
+  ASN1NumericString(this.stringValue, {int tag = NUMERIC_STRING_TYPE})
+      : super(tag: tag);
+
+  ///
+  /// Create an [ASN1PrintableString] from an encoded list of bytes.
```
Comment (wstrange) [inline lib/asn1numericstring.dart:22]: PrintableString or NumericString? Just checking

File: lib/asn1numericstring.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class ASN1NumericString extends ASN1Object {
+  // The decoded string value
+  late String stringValue;
+
+  ///
+  /// Create an [ASN1NumericString] initialized with String value.
+  ///
+  /// Optionally override the tag
+  ///
+  ASN1NumericString(this.stringValue, {int tag = NUMERIC_STRING_TYPE})
```
Comment (wstrange) [inline lib/asn1numericstring.dart:18]: Is it worthwhile doing a quick check to see if the string contains any non numeric/space chars?  I dont think it is required - so its good as is - but would be an option.

File: test/asn1numericstring_test.dart [hunk 3]
```diff
@@ -0,0 +1,91 @@
+library asn1test;
```
Comment (wstrange) [inline test/asn1numericstring_test.dart:1]: Thanks for including tests!

**Review by wstrange — COMMENTED** [review]
Thanks for the PR!

Can you bump the minor version in pub, and update the changelog. Then I will publish the new version

File: lib/asn1numericstring.dart (hunk 1 above)
Comment (cyberbobs) [reply lib/asn1numericstring.dart]: It was just me being inattentive to the copy-pasted comment. Good catch!

File: lib/asn1numericstring.dart (hunk 2 above)
Comment (cyberbobs) [reply lib/asn1numericstring.dart:18]: I've added a simple RegExp based parameter validation and a test for it with a separate commit.

File: test/asn1numericstring_test.dart (hunk 3 above)
Comment (cyberbobs) [reply test/asn1numericstring_test.dart:1]: No problem. I wouldn't know whether this works or not without it myself :)

**Comment by cyberbobs** [comment]
Thank you for the quick reply. I've fixed all the issues with this PR you mentioned.

**Comment by wstrange** [comment]
Thanks !!
I'll push to pub shortly

**Pull request merged**